FinSearchUnified/BusinessLogic/Models/FindologicArticleModel.php

```diff
@@ -406,18 +406,23 @@ protected static function checkIfHasValue($value)
 
     protected function setImages()
     {
-        $articleMainImages = $this->baseArticle->getImages()->getValues();
-        $mediaService = Shopware()->Container()->get('shopware_media.media_service');
-        $baseLink = Shopware()->Modules()->Core()->sRewriteLink();
         $imagesArray = [];
-        $replacements = [
-            '[' => '%5B',
-            ']' => '%5D'
-        ];
+        $articleMainImages = [];
+        $replacements = [];
+        $baseLink = Shopware()->Modules()->Core()->sRewriteLink();
+        $mediaService = Shopware()->Container()->get('shopware_media.media_service');
+
+        if ($this->baseArticle->getImages() !== null) {
+            $articleMainImages = $this->baseArticle->getImages()->getValues();
+            $replacements = [
```

Comment:
I think you could declare `$replacements` with these values above. They are static and won't change.